src/test/test-os-util.c

```diff
@@ -72,4 +72,18 @@ TEST(load_os_release_pairs) {
         assert_se(unsetenv("SYSTEMD_OS_RELEASE") == 0);
 }
 
+TEST(os_release_support_ended) {
+        int r;
+
+        assert_se(os_release_support_ended("1999-01-01", false) == true);
+        assert_se(os_release_support_ended("2090-01-01", false) == false);
```

Comment:
Surely, systemd will be obsolete by then… :joy_cat:

Reply:
Damn, I had to change this to 2037 because of i686. We'll need all to look for a new hobby soon ;)

Reply:
Wait, we aren't using 64bit versions of time APIs? O_O

Reply:
IIRC for some i386 distros that transition is not yet done, because it's ABI-breaking. I should probably book January 2038 as vacation in advance...